### PR TITLE
Holiday System

### LINKED ISSUE #3113: Holiday System
A system that keeps track of IRL holidays for the purpose of fun events and interactions on certain holidays.

### PR DESCRIPTION
**Pull request by Zumorica**
A whole hecking holiday system!
Includes over 55 holidays!

Based on Beestation and /vg/station's holiday system and holidays. 

Closes #3113

maybe TODO:
- [ ] Document more stuff?
- [ ] Add an example `IHolidayCelebration` for a holiday?

### CODE REVIEW

File: Resources/Prototypes/holidays.yml [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+  id: NewYear
+  name: New Year
+  beginDay: 30
+  beginMonth: December
+  endDay: 2
+  endMonth: January
+
+- type: holiday
+  id: MisterLizard
+  name: Mister Lizard's Birthday
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:11]: image

Comment (RemieRichards) [reply Resources/Prototypes/holidays.yml:11]: swept

**Comment by SweptWasTaken** [comment]
What do the holidays actually do. Also might just be me but seems like there's a ton of holiday bloat.

**Comment by RemieRichards** [comment]
@SweptWasTaken they greet you.
they can also run code (i.e. spawn christmas trees around the station)

**Comment by Zumorica** [comment]
> Also might just be me but seems like there's a ton of holiday bloat.

Most of the holidays are taken straight from bee and /vg/, I only added like two or three

File: Resources/Prototypes/holidays.yml [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+- type: holiday
+  id: SummerSolstice
+  name: Summer Solstice
+  beginDay: 21
+  beginMonth: June
+
+- type: holiday
+  id: StonewallRiotsAnniversary
+  name: Stonewall Riots Anniversary
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:162]: Can we replace this with "Spirit Day"

https://www.glaad.org/spiritday

Celebrated it at my school in US when I was a kid, it's on October 17th

File: Resources/Prototypes/holidays.yml [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+
+- type: holiday
+  id: UFODay
+  name: UFO Day
+  beginDay: 2
+  beginMonth: July
+
+- type: holiday
+  id: IndependenceDay
+  name: Independence Day
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:180]: Should specify this as "US"

File: Resources/Prototypes/holidays.yml [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+
+- type: holiday
+  id: WritersDay
+  name: Writer's Day
+  beginDay: 8
+  beginMonth: July
+
+- type: holiday
+  id: BastilleDay
+  name: Bastille Day
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:192]: Eh, I would scrap this one.

File: Resources/Prototypes/holidays.yml [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+
+- type: holiday
+  id: AnimalsDay
+  name: Animal's Day
+  beginDay: 4
+  beginMonth: October
+
+- type: holiday
+  id: SmilingDay
+  name: Smiling Day
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:243]: Eh

File: Resources/Prototypes/holidays.yml [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+- type: holiday
+  id: CanadianThanksgiving
+  name: Thanksgiving in Canada
+  beginWeek: 2
+  beginWeekday: Monday
+  beginMonth: October
+
+- type: holiday
+  id: BossDay
+  name: Boss' Day
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:262]: Eh

File: Resources/Prototypes/holidays.yml (hunk 3 above)
Comment (Zumorica) [reply Resources/Prototypes/holidays.yml]: Done

File: Resources/Prototypes/holidays.yml [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+
+- type: holiday
+  id: TransgenderRemembranceDay
+  name: Transgender Day of Remembrance
+  beginDay: 20
+  beginMonth: November
+
+- type: holiday
+  id: SayingHelloDay
+  name: Saying Hello Day
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:310]: Scrap

File: Resources/Prototypes/holidays.yml [hunk 8]
```diff
@@ -0,0 +1,369 @@
...
+- type: holiday
+  id: FestiveSeason
+  name: Festive Season
+  beginDay: 1
+  endDay: 31
+  beginMonth: December
+
+- type: holiday
+  id: BoxingDay
+  name: Boxing Day
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:361]: Only really celebrated in UK, falls under Christmas?

File: Resources/Prototypes/holidays.yml [hunk 9]
```diff
@@ -0,0 +1,369 @@
...
+
+- type: holiday
+  id: CosmonauticsDay
+  name: Cosmonautics Day
+  beginDay: 12
+  beginMonth: April
+
+- type: holiday
+  id: SalyutAnniversary
+  name: Salyut 1's Launch Anniversary
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:84]: I'd group this under Cosmonautics

File: Resources/Prototypes/holidays.yml [hunk 10]
```diff
@@ -0,0 +1,369 @@
...
+  name: Mother's Day
+  beginMonth: May
+  shouldCelebrate:
+    !type:WeekdayInMonth
+    weekday: Sunday
+    occurrence: 2
+
+- type: holiday
+  id: OwlAndPussycatDay
+  name: Owl and Pussycat Day
```
Comment (SweptWasTaken) [inline Resources/Prototypes/holidays.yml:135]: Super niche, i'd scrap

**Comment by SweptWasTaken** [comment]
Added some reviews, probably super subjective and there's no reason "not" to have them so it's up to you vera

File: Resources/Prototypes/holidays.yml (hunk 8 above)
Comment (RemieRichards) [reply Resources/Prototypes/holidays.yml:361]: don't you dare touch boxing day.

if you're having US Independence day, we're having boxing day.

**Comment by ike709** [comment]
> No Garbage Day

How is the Janitor supposed to get a gun???

**Comment by Zumorica** [comment]
> > Garbage Day not included
> 
> How is the Janitor supposed to get a gun???

oops, I excluded this one by mistake. Will add now

File: Resources/Prototypes/holidays.yml (hunk 8 above)
Comment (SweptWasTaken) [reply Resources/Prototypes/holidays.yml:373]: > 
> 
> don't you dare touch boxing day.
> 
> if you're having US Independence day, we're having boxing day.

Should we even have US Independence day?

Comment (RemieRichards) [reply Resources/Prototypes/holidays.yml:373]: no.

but we're still having boxing day.

:3c

Comment (metalgearsloth) [reply Resources/Prototypes/holidays.yml:373]: Straya also has it.

**Comment by metalgearsloth** [comment]
Could add lunar new year (melbourne has a large chinese population cough*)

**Pull request merged**